homeassistant/config.py

```diff
@@ -824,11 +830,16 @@ async def merge_packages_config(
 
 
 async def async_process_component_config(  # noqa: C901
-    hass: HomeAssistant, config: ConfigType, integration: Integration
+    hass: HomeAssistant,
+    config: ConfigType,
+    integration: Integration,
+    raise_on_failure: bool = False,
```

Comment:
Make this a kw-only arg
```suggestion
    *,
    raise_on_failure: bool = False,
```